internal/hooks/v0hooks/manager.go

```diff
@@ -188,47 +97,116 @@ func (o *Manager) runHook(
 	return nil
 }
 
-func checkError(
-	hookOutput HookOutput,
+func configByName(
+	cfg *conf.HookConfiguration,
+	name Name,
+) (*conf.ExtensibilityPointConfiguration, bool) {
+	switch name {
+	case SendSMS:
+		return &cfg.SendSMS, true
+	case SendEmail:
+		return &cfg.SendEmail, true
+	case CustomizeAccessToken:
+		return &cfg.CustomAccessToken, true
+	case MFAVerification:
+		return &cfg.MFAVerificationAttempt, true
+	case PasswordVerification:
+		return &cfg.PasswordVerificationAttempt, true
+	case BeforeUserCreated:
+		return &cfg.BeforeUserCreated, true
+	case AfterUserCreated:
+		return &cfg.AfterUserCreated, true
+	default:
+		return nil, false
+	}
+}
+
+func (o *Manager) Enabled(name Name) bool {
+	if cfg, ok := configByName(&o.config.Hook, name); ok {
+		return cfg.Enabled
+	}
+	return false
+}
+
+func (o *Manager) BeforeUserCreated(
+	ctx context.Context,
+	tx *storage.Connection,
+	req *BeforeUserCreatedRequest,
+	res *BeforeUserCreatedResponse,
 ) error {
-	if hookOutput.IsError() {
-		he := hookOutput.GetHookError()
-		httpCode := he.HTTPCode
+	return o.dr.Dispatch(ctx, &o.config.Hook.BeforeUserCreated, tx, req, res)
+}
 
-		if httpCode == 0 {
-			httpCode = http.StatusInternalServerError
-		}
+func (o *Manager) AfterUserCreated(
+	ctx context.Context,
+	tx *storage.Connection,
+	req *AfterUserCreatedRequest,
+	res *AfterUserCreatedResponse,
+) error {
+	return o.dr.Dispatch(ctx, &o.config.Hook.AfterUserCreated, tx, req, res)
+}
```

Comment:
Why are `BeforeUserCreated` and `AfterUserCreated` using `...Request` and `...Response` while the existing hooks like `CustomAccessToken` use `...Input` and `...Output`.

This is one more unnecessary distinction that does not help the readability or comprehension of the code. Both after user created and before user created use the same underlying concept of sending an **input** to the hook handler (Postgres function or over HTTP), and receive an **output.** Postgres functions can't do "requests" so the minimal subset is just dealing with the payload, and the transport layer depends on the configuration.

Reply:
sure, done